comments-bundle/src/EventListener/DataContainer/CommentsConditionalFieldsListener.php

```diff
@@ -0,0 +1,176 @@
+            'inputType' => 'checkbox',
+            'eval' => ['tl_class' => 'w50'],
+            'sql' => ['type' => 'boolean', 'default' => false],
+        ];
+
+        switch ($table) {
+            case 'tl_news':
+            case 'tl_calendar_events':
+                PaletteManipulator::create()
+                    ->addLegend('comments_legend', 'protected_legend', PaletteManipulator::POSITION_AFTER, true)
+                    ->addField(['allowComments'], 'comments_legend', PaletteManipulator::POSITION_APPEND)
+                    ->applyToPalette('default', $table)
+                ;
+
+                break;
+
+            case 'tl_faq':
+                PaletteManipulator::create()
+                    ->addLegend('comments_legend', 'title_legend', PaletteManipulator::POSITION_AFTER, true)
+                    ->addField(['allowComments'], 'comments_legend', PaletteManipulator::POSITION_APPEND)
+                    ->applyToPalette('default', $table)
+                ;
+
+                break;
+        }
+    }
+
+    private function applyChildrenFields(string $table): void
+    {
+        $GLOBALS['TL_DCA'][$table]['list']['sorting']['headerFields'][] = 'allowComments';
+
+        switch ($table) {
+            case 'tl_news':
+                $GLOBALS['TL_DCA'][$table]['fields']['noComments'] = [
+                    'filter' => true,
+                    'inputType' => 'checkbox',
+                    'eval' => ['tl_class' => 'w50 m12'],
+                    'sql' => ['type' => 'boolean', 'default' => false],
+                ];
+                break;
+
+            case 'tl_calendar_events':
+                $GLOBALS['TL_DCA'][$table]['fields']['noComments'] = [
+                    'inputType' => 'checkbox',
+                    'eval' => ['tl_class' => 'w50 m12'],
+                    'sql' => ['type' => 'boolean', 'default' => false],
+                ];
+                break;
+
+            case 'tl_faq':
+                $GLOBALS['TL_DCA'][$table]['fields']['noComments'] = [
+                    'filter' => true,
+                    'inputType' => 'checkbox',
+                    'sql' => ['type' => 'boolean', 'default' => false],
+                ];
+                break;
+        }
+
+        switch ($table) {
+            case 'tl_news':
+            case 'tl_calendar_events':
+                PaletteManipulator::create()
+                    ->addField(['noComments'], 'expert_legend', PaletteManipulator::POSITION_APPEND)
+                    ->applyToPalette('default', $table)
+                    ->applyToPalette('internal', $table)
+                    ->applyToPalette('article', $table)
+                    ->applyToPalette('external', $table)
+                ;
+
+                break;
+
+            case 'tl_faq':
+                PaletteManipulator::create()
+                    ->addLegend('expert_legend', 'publish_legend', PaletteManipulator::POSITION_BEFORE, true)
```

Comment:
if the legend already exists, the palette manipulator should simply ignore this (or maybe change the order 🤔). So maybe we could apply the same call to all tables?

Reply:
The expert legend does not exist in the faq table, hence why I am handling most of it this way. We could add it within the faq-bundle without a field so it won't be shown, wdyt?

Reply:
That's not what I mean. If you use this code in the news and calendar bundle as well, it should simply do nothing because the legend already exists. So no switch-case would be necessary.

Reply:
I checked this again, the problem ist that news and calendar_events have 3 more palettes that this option has to be added:
```php
PaletteManipulator::create()
    ->addField(['noComments'], 'expert_legend', PaletteManipulator::POSITION_APPEND)
    ->applyToPalette('default', $table)
    ->applyToPalette('internal', $table)
    ->applyToPalette('article', $table)
    ->applyToPalette('external', $table)
;
```

Whilst the faq bundle only has one palette (default). The only way to circumvent this would be changing the PaletteNotFoundException to not be thrown 🤔 ?